flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/functions/DeclaringAsyncKeyedCoProcessFunction.java

```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * This method is called for each element in the first of the connected streams.
+     *
+     * <p>This function can output zero or more elements using the {@link Collector} parameter and
+     * also update internal state or set timers using the {@link Context} parameter.
+     *
+     * @param value The stream element
+     * @param ctx A {@link Context} that allows querying the timestamp of the element, querying the
+     *     {@link TimeDomain} of the firing timer and getting a {@link TimerService} for registering
+     *     timers and querying the time. The context is only valid during the invocation of this
+     *     method, do not store it.
+     * @param out The collector to emit resulting elements to
+     * @throws Exception The function may throw exceptions which cause the streaming program to fail
+     *     and go into recovery.
+     */
+    public void processElement1(IN1 value, Context ctx, Collector<OUT> out) throws Exception {
+        throw new IllegalAccessException("This method is replaced by declareProcess1.");
+    }
+
+    /**
+     * This method is called for each element in the second of the connected streams.
+     *
+     * <p>This function can output zero or more elements using the {@link Collector} parameter and
+     * also update internal state or set timers using the {@link Context} parameter.
+     *
+     * @param value The stream element
+     * @param ctx A {@link Context} that allows querying the timestamp of the element, querying the
+     *     {@link TimeDomain} of the firing timer and getting a {@link TimerService} for registering
+     *     timers and querying the time. The context is only valid during the invocation of this
+     *     method, do not store it.
+     * @param out The collector to emit resulting elements to
+     * @throws Exception The function may throw exceptions which cause the streaming program to fail
+     *     and go into recovery.
+     */
+    public void processElement2(IN2 value, Context ctx, Collector<OUT> out) throws Exception {
+        throw new IllegalAccessException("This method is replaced by declareProcess2.");
+    }
+
+    /**
+     * Called when a timer set using {@link TimerService} fires.
+     *
+     * @param timestamp The timestamp of the firing timer.
+     * @param ctx An {@link OnTimerContext} that allows querying the timestamp of the firing timer,
+     *     querying the {@link TimeDomain} of the firing timer and getting a {@link TimerService}
+     *     for registering timers and querying the time. The context is only valid during the
+     *     invocation of this method, do not store it.
+     * @param out The collector for returning result values.
+     * @throws Exception This method may throw exceptions. Throwing an exception will cause the
+     *     operation to fail and may trigger recovery.
+     */
+    public void onTimer(long timestamp, OnTimerContext ctx, Collector<OUT> out) throws Exception {}
+
+    /**
```

Comment:
this javadoc has no associated code.

Reply:
It should be `/**`, because it is for a method.

Reply:
@Zakelly `KeyedCoProcessFunction#Context` isn't removed, I added `AsyncKeyedCoProcessOperator#ContextImpl` which inherits from it.